modules/openideal_login/openideal_login.module

```diff
@@ -0,0 +1,194 @@
+  foreach ($social_plugins as $plugin_id => &$data) {
+    if (!$network_manager->createInstance($plugin_id)->getSdk()) {
+      unset($social_plugins[$plugin_id]);
+    }
+
+    $theme_path = \Drupal::service('theme_handler')->getTheme('openideal_theme')->getPath();
+    $prefix = 'Sign in with';
+    if (strpos($plugin_id, 'google') !== FALSE) {
+      $data['img_path'] = $theme_path . '/images/icons/google_icon.png';
+      $data['name'] = t('@prefix Google', ['@prefix' => $prefix]);
+    }
+    elseif (strpos($plugin_id, 'facebook') !== FALSE) {
+      $data['img_path'] = $theme_path . '/images/icons/facebook_icon.png';
+      $data['name'] = t('@prefix Facebook', ['@prefix' => $prefix]);
+    }
+    elseif (strpos($plugin_id, 'github') !== FALSE) {
+      $data['img_path'] = $theme_path . '/images/icons/github_icon.png';
+      $data['name'] = t('@prefix GitHub', ['@prefix' => $prefix]);
+    }
+  }
+
+  if (!empty($social_plugins)) {
+    $form['container']['left']['or'] = [
+      '#type' => 'container',
+      '#attributes' => ['class' => ['user-login-form--container--left__or']],
+      'before' => [
+        '#type' => 'container',
+        '#attributes' => ['class' => ['user-login-form--container--left__or--line']],
+      ],
+      'or' => [
+        '#type' => 'html_tag',
+        '#tag' => 'p',
+        '#value' => 'or',
+        '#attributes' => ['class' => ['user-login-form--container--left__or--or']],
+      ],
+      'after' => [
+        '#type' => 'container',
+        '#attributes' => ['class' => ['user-login-form--container--left__or--line']],
+      ],
+      '#weight' => '2',
+    ];
+
+    $form['container']['left']['social_buttons'] = [
+      '#type' => 'container',
+      '#attributes' => ['class' => ['user-login-form--container--left__social-login-buttons']],
+      'buttons' => [
+        '#theme' => 'login_with',
+        '#social_networks' => $social_plugins,
+        '#weight' => '1',
+      ],
+    ];
+  }
+
+  $signup_link = Link::createFromRoute(t('signup'), 'user.register')->toString();
+  $reset_link = Link::createFromRoute(t('Forgot your password?'), 'user.pass')->toString();
+  $form['container']['left']['bottom'] = [
+    '#type' => 'container',
+    '#attributes' => ['class' => ['user-login-form--container__bottom']],
+    'signup' => [
+      '#type' => 'html_tag',
+      '#attributes' => [
+        'class' => [
+          'user-login-form--container__bottom--signup',
+          'user-login-form--container__bottom--text',
+        ],
+      ],
+      '#tag' => 'p',
+      '#value' => t("Don't have an account? @link", ['@link' => $signup_link]),
+    ],
+    'reset' => [
+      '#type' => 'html_tag',
+      '#attributes' => [
+        'class' => [
+          'user-login-form--container__bottom--forgot-password',
+          'user-login-form--container__bottom--text',
+        ],
+      ],
+      '#tag' => 'p',
+      '#value' => t("@link", ['@link' => $reset_link]),
```

Comment:
What's the purpose of this t() function call?

Reply:
Done..